CROSS_PLATFORM_BUILD_GUIDE.md

```
@@ -0,0 +1,150 @@
+The build system handles these mappings automatically.
+
+## Troubleshooting
+
+### Build works without explicit architecture
+
+The build system now auto-detects your system architecture if BuildKit is not enabled. However, for explicit multi-architecture builds, you should use BuildKit with the `--platform` flag.
+
+### Error: Unsupported architecture
+
+**Symptom:**
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)